### PR TITLE
[pytorch] correct input size check for GroupNorm

### PR DESCRIPTION
**Pull request by vkhalidov**
Summary:
Corrects D19373507 to allow valid use cases that fail now. Multiplies batch size by the number of elements in a group to get the correct number of elements over which statistics are computed.

**Details**:
The current implementation disallows GroupNorm to be applied to tensors of shape e.g. `(1, C, 1, 1)` to prevent cases where statistics are computed over 1 element and thus result in a tensor filled with zeros.
However, in GroupNorm the statistics are calculated across channels. So in case where one has an input tensor of shape `(1, 256, 1, 1)` for `GroupNorm(32, 256)`, the statistics will be computed over 8 elements and thus be meaningful.

One use case is [Atrous Spatial Pyramid Pooling (ASPPPooling)](https://github.com/pytorch/vision/blob/791c172a337d98012018f98ffde93b1020ba3ed5/torchvision/models/segmentation/deeplabv3.py#L50), where GroupNorm could be used in place of BatchNorm [here](https://github.com/pytorch/vision/blob/791c172a337d98012018f98ffde93b1020ba3ed5/torchvision/models/segmentation/deeplabv3.py#L55). However, now this is prohibited and results in failures.

Proposed solution consists in correcting the computation of the number of elements over which statistics are computed. The number of elements per group is taken into account in the batch size.

Test Plan: check that existing tests pass

Differential Revision: D19723407

### CODE REVIEW

**Comment by ezyang** [comment]
I'm not really the right reviewer for this, but this needs a test at least.

**Comment by ezyang** [comment]
@v0dro, you added the checks, could you please take a look at this? Thanks!

**Comment by v0dro** [comment]
@ezyang the reason given for the change seems legit to me.
Also none of the previous tests break so I think there's no problem with this change.

**Review by v0dro — COMMENTED** [review]
Looks good to me. The reason given for the changes seems legit.

**Review by ezyang — APPROVED** [review]
